Sources/SwiftDocC/Infrastructure/Symbol Graph/SymbolGraphLoader.swift

```diff
@@ -201,130 +207,130 @@ struct SymbolGraphLoader {
-            registeredPlatforms.contains($0.value)
-        }
-        // Platforms that are defined in the Info.plist that had no corresponding SGF
-        // and are not being added as fallback of another platform.
-        let missingAvailabilities = defaultAvailabilities.filter {
-            !missingFallbackPlatforms.keys.contains($0.platformName) &&
-            !registeredPlatforms.contains($0.platformName)
-        }
-        
-        unifiedGraph.symbols.values.forEach { symbol in
+        unifiedGraph.symbols.forEach { (symbolID, symbol) in
+            
+            // The fallback platforms that are not marked as unavailable in the default availability.
+            let fallbackPlatforms = DefaultAvailability.fallbackPlatforms.filter {
+                !unconditionallyUnavailablePlatformNames.contains($0.key)
+            }
+            
+            // The symbol availability grouped by interface language.
+            // Is needed to separate the availability in language sets because the symbol availability
+            // can difer betwen language variants.
+            let symbolAvailabilityByInterfaceLanguage = Dictionary(grouping: symbol.availability, by: { $0.key.interfaceLanguage })
+            
             for (selector, _) in symbol.mixins {
-                if var symbolAvailability = (symbol.mixins[selector]?["availability"] as? SymbolGraph.Symbol.Availability) {
-                    guard !symbolAvailability.availability.isEmpty else { continue }
-                    // For platforms with a fallback option (e.g., Catalyst and iOS), apply the explicit availability annotation of the fallback platform when it is not explicitly available on the primary platform.
-                    DefaultAvailability.fallbackPlatforms.forEach { (fallbackPlatform, inheritedPlatform) in
+                // Platforms available for the given symbol in the given language variant.
+                var platformsAvailableAtThisSymbol = symbolAvailabilityByInterfaceLanguage[selector.interfaceLanguage].flatMap {
+                    $0.flatMap { symbolAvailability in
+                        symbolAvailability.value.filter({ item in
+                            item.introducedVersion != nil
+                        }).map(\.domain?.rawValue)
+                    }
+                } ?? []
+                
+                // The symbol availability for the given selector.
+                var symbolAvailability = (symbol.mixins[selector]?[SymbolGraph.Symbol.Availability.mixinKey] as? SymbolGraph.Symbol.Availability) ?? SymbolGraph.Symbol.Availability(availability: [])
+                
+                // For platforms with an inherited platform, add the fallback platform if this one
+                // does not exists already in the registered platforms available in this language.
+                if !symbolAvailability.availability.isEmpty {
+                    fallbackPlatforms.forEach { (fallbackPlatform, inheritedPlatform) in
```

Comment:
My main thought when I look at this loop is that we're repeating work for every symbol for the same platform to inherit by. 

Right now the fallback platforms is defined like
```swift
[
  .catalyst: .iOS,
  .iPadOS:   .iOS,
]
```

However, if the data was grouped by the platform to inherit by
```swift
[ .iOS: [.catalyst, .iPadOS]]
```
then it might be possible to make the computation once and apply them to both fallback platforms.

Reply:
Resolved here https://github.com/swiftlang/swift-docc/pull/1065/commits/8b2cb42e45c6fc2369f46ddc7ca2fae9094808c3